### PR TITLE
[Cluster Agent] Support internal profiling

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

Add internal profiling support to the cluster agent

### Motivation

Better observability

### Additional Notes

Can be configured at runtime (similar to the agent)

### Describe how to test/QA your changes

Enable `internal_profiling.enabled` + configure `DD_AGENT_HOST`

image

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/cluster-agent/app/app.go [hunk 1]
```diff
@@ -157,6 +158,26 @@ func start(cmd *cobra.Command, args []string) error {
...
+		}
+	}
+
+	if config.Datadog.GetBool("internal_profiling.enabled") {
+		err := settings.SetRuntimeSetting("internal_profiling", true)
+		if err != nil {
+			log.Errorf("Error starting profiler: %v", err)
+		}
+	}
+
```
Comment (clamoriniere) [inline cmd/cluster-agent/app/app.go:180]: I'm wondering if we should create a shared function with other agents.

Comment (clamoriniere) [reply cmd/cluster-agent/app/app.go:180]: we have it in several place, but never all the same options....
* https://github.com/DataDog/datadog-agent/blob/9925e508eec861aa809668e2dcf145bacda27174/cmd/system-probe/config/config.go#L132
* https://github.com/DataDog/datadog-agent/blob/45c01f4bab71f397497edaee2f662712eabd04b0/cmd/trace-agent/run.go#L276
* https://github.com/DataDog/datadog-agent/blob/92d7c61fa33ecebc4575027f0b20cf630fcf029e/cmd/process-agent/main_common.go#L219

Comment (ahmed-mez) [reply cmd/cluster-agent/app/app.go]: I added a shared function between the cluster agent and the core agent.  But process/trace/system-probe are building their own profiling config so it doesn't make sense to include them imho.

**Review by vboulineau — APPROVED** [review]

**Review by djmitche — APPROVED** [review]

**Review by jtappa — APPROVED** [review]
👍🏻 from docs for release note

**Comment by remeh** [comment]
Confidence check: this change means we'll use the same configuration to enable the internal profiling of the regular agent or of the cluster agent, there is no scenario where we would need to enable it through configuration / a chart on one and not the another and this change would block us to do so?

**Comment by vboulineau** [comment]
> Confidence check: this change means we'll use the same configuration to enable the internal profiling of the regular agent or of the cluster agent, there is no scenario where we would need to enable it through configuration / a chart on one and not the another and this change would block us to do so?

The configuration **parameters** are the same, but the DCA and the Agent do not share the same `datadog.yaml` or the same environment variables.

**Comment by remeh** [comment]
@vboulineau @ahmed-mez perfect, thank you!

**Pull request merged**